packages/data-context/src/actions/ElectronActions.ts

```diff
@@ -1,6 +1,7 @@
 import type { BrowserWindow } from 'electron'
 import os from 'os'
 import type { DataContext } from '..'
+import { openExternal } from '@packages/server/lib/gui/links'
 
```

Comment:
Sorry I missed this in the first review.

One pattern we are following is trying to keep the data-context layer decoupled from the server. You will notice if you search the entire `data-context` package, there are 0 imports from server.

I don't think we should be importing this here. Rather, we should inject it when we call `makeDataContext` in the server. See here: https://github.com/cypress-io/cypress/blob/unified-desktop-gui/packages/server/lib/makeDataContext.ts#L51

I think we should do something like this:

```ts
export function makeDataContext (options: MakeDataContextOptions) {
  return new DataContext({
    schema: graphqlSchema,
    ...options,
    // ...
    electronApi: {
      openExternal (url: string) {
        openExternal(url)
      }
    }
```

Then in `ElectronActions.ts`:

```ts
  openExternal (url: string) {
    this.ctx.electronApi.openExternal(url)
  }
```

There is a similar pattern on line 35 of this file:

https://github.com/cypress-io/cypress/pull/18780/files#diff-ec23b134998d4539735d9d91ff7585b09845345eb0661feb222b90c24bac06f6R35

```ts
showBrowserWindow () {
    this.electron.browserWindow?.show()
    if (this.isMac) {
      this.ctx.electronApp.dock.show()
    } else {
      this.electron.browserWindow?.setSkipTaskbar(false)
    }
  }
```